fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java

```diff
@@ -1567,10 +1570,42 @@ private void handleShowData() throws AnalysisException {
 
     private void handleShowPartitions() throws AnalysisException {
         ShowPartitionsStmt showStmt = (ShowPartitionsStmt) stmt;
-        ProcNodeInterface procNodeI = showStmt.getNode();
-        Preconditions.checkNotNull(procNodeI);
-        List<List<String>> rows = ((PartitionsProcDir) procNodeI).fetchResultByFilter(showStmt.getFilterMap(),
-                showStmt.getOrderByPairs(), showStmt.getLimitElement()).getRows();
+        if (showStmt.getCatalog().isInternalCatalog()) {
+            ProcNodeInterface procNodeI = showStmt.getNode();
+            Preconditions.checkNotNull(procNodeI);
+            List<List<String>> rows = ((PartitionsProcDir) procNodeI).fetchResultByFilter(showStmt.getFilterMap(),
+                    showStmt.getOrderByPairs(), showStmt.getLimitElement()).getRows();
+            resultSet = new ShowResultSet(showStmt.getMetaData(), rows);
+        } else {
+            handleShowHMSTablePartitions(showStmt);
+        }
+    }
+
+    private void handleShowHMSTablePartitions(ShowPartitionsStmt showStmt) throws AnalysisException {
+        CatalogIf catalog = showStmt.getCatalog();
+        DatabaseIf database = catalog.getDbOrAnalysisException(showStmt.getTableName().getDb());
+        HMSExternalTable hmsTable = (HMSExternalTable) database.getTableNullable(showStmt.getTableName().getTbl());
+        List<Type> partitionColumnTypes = hmsTable.getPartitionColumnTypes();
+
+        List<List<String>> rows = new ArrayList<>();
+        HiveMetaStoreCache cache = Env.getCurrentEnv().getExtMetaCacheMgr()
+                .getMetaStoreCache((HMSExternalCatalog) catalog);
+        if (!partitionColumnTypes.isEmpty()) {
+            HiveMetaStoreCache.HivePartitionValues hivePartitionValues =
+                    cache.getPartitionValues(hmsTable.getDbName(), hmsTable.getName(), partitionColumnTypes);
```

Comment:
If there are too many partitions, this operation will refresh the entire cache and may impact other query.
I suggest that for `show partitions` operation, not using cache, but to get partition directly from hms.
And we also need to consider if partition num is too large, eg, 10 thousands, do we need to limit the num of returned rows?

Reply:
fixed